### ISSUE TITLE
TypeScript support

### ISSUE THREAD

**Issue by muratkeremozcan** [bahmutov/cy-spok#9]
Trying to make cy-spok work with typescript. 

Making a plugin work with Typescript is usually not easy, but I tried all ways and still am failing.

Does cy-spok work with Typescript, is there an example we can look at?

**Comment by Parakoos** [comment]
Hi. I'm pretty new to Cypress so this is probably all wrong, but, here is what I did to make cy-spok a bit more user friendly and type-safe.

```typescript
// index.d.ts
type SpecHelper = {
  ge: (arg: number) => SatisfactionChecker<number>
  notDefined: SatisfactionChecker<undefined | null>
  // This can be extended with all the different methhods avaiable at https://github.com/thlorenz/spok#api
}
type SatisfactionChecker<T> = (input: T) => boolean
type SpecGenerator = (helper: SpecHelper) => Spec
type Spec = { [key: string]: any }

declare namespace Cypress {
  interface Chainable<Subject> {
    /**
     * Takes the subject checks it against cy-spok to verify that is satisfies the given specification.
     * The specification can either be a static specification or a function that creates one just before
     * the 'should' test is performed. If the latter function is used, it will be given a cy-spok instance
     * as a helper argument (to create satsifaction helper functions) and a 'this' context of the test
     * environment (including access to values saves using .as('ref') by calling this.ref).
     *
     * @param specification Either a static specification or a function that creates one dynamically.
     */
    shouldSatisfy(specification: SpecGenerator | Spec): Chainable<Subject>
  }
}
```

Then the implementation...
```javascript
// support.js
import spok from 'cy-spok'
Cypress.Commands.add('shouldSatisfy', { prevSubject: true }, (subject, specificationOrFn) => {
  const spec = typeof specificationOrFn === 'function' ? specificationOrFn(spok) : specificationOrFn
  cy.wrap(subject, { log: false }).should(spok(spec))
})
````

Finally, as used in a test
```javascript
cy.wrap(o).shouldSatisfy({
  $topic: 'spok-example-static',
  one: 1,
})

cy.wrap(o).shouldSatisfy(spok => ({
  $topic: 'spok-example-dynamic',
  one: spok.ge(1), // now with auto-complete!
  savedRef: this.someRefSavedPreviouslyUsingCyAs
}))
```

I'm sure this can be improved in many many ways, and if there are serious issues with the approach, I'd love to hear about it. I cobbled this together in a morning so use at own risk! :-)

**Comment by Stumblinbear** [comment]
Created a PR for typescript definitions: #68

**Comment by bahmutov** [comment]
Solved by #68 released in 1.5.1
